### PR TITLE
LPS-166661 LPS-167172 UI improvements semantic search

### PR DESCRIPTION
**Pull request by oliv-yu**
https://issues.liferay.com/browse/LPS-167172 - [FE] UI improvements for semantic search settings form

- Small bug when you save `0` in number inputs, the value is seen as falsey. The conditionals inside the validation were updated to check for just an empty string.
- Also updated some of the one-liner functions, and got rid of some inline functions.
- Sentence Transform Provider was given a default value (probably will never be an issue, but if it were nothing it doesn't render its corresponding inputs)

### CODE REVIEW

**Comment by oliv-yu** [comment]
ci:test:search

**Comment by oliv-yu** [comment]
ci:stop

**Comment by oliv-yu** [comment]
ci:test:sf

**Comment by oliv-yu** [comment]
ci:test:search

File: modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/semantic_search/js/configuration/ModelAutocomplete.js [hunk 1]
```diff
@@ -19,43 +20,107 @@ import React, {useState} from 'react';
...
+
+	const _handleInputChange = (event) => {
+		setShowDropDown(true);
+		setAutocompleteSearchValue(event.target.value);
+	};
+
+	const _handleItemChange = (item) => {
+		setAutocompleteSearchValue(item);
+		onChange(item);
+		setShowDropDown(false);
```
Comment (thektan) [inline modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/semantic_search/js/configuration/ModelAutocomplete.js:82]: If lines are grouped together, they should be sorted. If they shouldn't be, use a blank line in between.

```
setAutocompleteSearchValue(event.target.value);
setShowDropDown(true);
```

**Review by thektan — CHANGES_REQUESTED** [review]
Can you make this small change? And then you can go ahead and run forward afterwards.

File: modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/semantic_search/js/configuration/ModelAutocomplete.js (hunk 1 above)
Comment (oliv-yu) [reply modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/semantic_search/js/configuration/ModelAutocomplete.js]: Added!

**Review by thektan — APPROVED** [review]

**Pull request closed without merging**